### PR TITLE
refactor: components

### PR DESCRIPTION
**Pull request by alex-mngt**
Refacto of all components to match the new folder architecture

### CODE REVIEW

File: src/components/Tag/Tag.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  className?: string
+  text: string
+  color?: ITagColor
+}
+
+export const Tag: FC<IProps> = ({ text, color = 'base-white', className }) => {
+  const classNames = getClassNames(color, className)
+
+  return (
+    <div className={classNames.join(' ')}>
```
Comment (moreirathomas) [inline src/components/Tag/Tag.tsx:18]: Maybe we can move the call to `join(' ')` inside the function here too ?

File: src/typing/colors.ts [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
-  httpPut = 'http-put',
-  httpPatch = 'http-patch',
-  httpDelete = 'http-delete',
-  grey = 'grey',
-  greyLight = 'grey-light',
-  greyDark = 'grey-dark',
-  primary = 'primary',
-  baseWhite = 'base-white',
-}
+export type IColors =
```
Comment (moreirathomas) [inline src/typing/colors.ts:1]: Nice a union string is better !

**Review by moreirathomas — APPROVED** [review]
Very nice ! A very small nitpick only :)

File: src/components/Button/internal/Button.helpers.ts [hunk 3]
```diff
@@ -2,15 +2,14 @@ import { createElement, ReactNode } from 'react'
...
-  small,
-  className,
-}: Pick<IProps, 'className'> &
-  Pick<Required<IProps>, 'color' | 'style' | 'small'>) => {
+export const getClassNames = (
+  color: IButtonColor,
+  style: IButtonStyle,
+  small: boolean,
+  className: string | undefined
+): string => {
```
[the remark this answers is not in the document]
Comment (alex-mngt) [reply src/components/Button/internal/Button.helpers.ts:12]: I did the change for simplifying the method signature but I do agree that it makes it harder to maintain. I'll switch back but only keep a list of params for methods which only take a single argument

**Review by nartydev — APPROVED** [review]

**Review by c1-ra — APPROVED** [review]

**Pull request merged**